File: 2444. Count Subarrays With Fixed Bounds.java
class Solution {
    public long countSubarrays(int[] nums, int minK, int maxK) {
        long answer = 0;
        int badI = -1;
        int minI = -1;
        int maxI = -1;
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] < minK || nums[i] > maxK) badI = i;
            if (nums[i] == minK) minI = i;
            if (nums[i] == maxK) maxI = i;
            answer += Math.max(0, Math.min(minI, maxI) - badI);
        }
        return answer;
    }
}

File: 2444. Count Subarrays With Fixed Bounds.py
class Solution:
    def countSubarrays(self, nums: List[int], minK: int, maxK: int) -> int:
        answer, minI, maxI, badI = 0, -1, -1, -1
        for i in range(len(nums)):
            if nums[i] < minK or nums[i] > maxK:
                badI = i
            if nums[i] == minK:
                minI = i
            if nums[i] == maxK:
                maxI = i
            answer += max(0, min(minI, maxI) - badI)
        return answer
            